src/test/ui/consts/const-eval/issue-62272.rs

```diff
@@ -1,9 +1,11 @@
-// run-pass
+// `loop`s unconditionally-broken-from used to be allowed in constants, but are now forbidden by
+// the HIR const-checker.
+//
+// See https://github.com/rust-lang/rust/pull/66170 and
+// https://github.com/rust-lang/rust/issues/62272.
 
-// Tests that `loop`s unconditionally-broken-from are allowed in constants.
-
-const FOO: () = loop { break; };
+const FOO: () = loop { break; }; //~ ERROR `loop` is not allowed in a `const`
```

Comment:
Since this is a breaking change for stable code I don't see how we can do this.

Reply:
- This has never been intentionally stabilized by the language team, as such, https://github.com/rust-lang/rfcs/pull/2405 applies.
- I checked with godbolt and this was an error in 1.37.0.
- Writing `loop { break; }` instead of `()` is just golfing; real code wouldn't do this.

Reply:
This worked on stable 1.21 through 1.24. It was then broken in all subsequent versions until 1.38, when it became allowed again.